File: Frontend/src/app/interfaces/UserAsFriendSearchResultDTO.ts
export interface UserAsFriendSearchResultDTO {
  id: number;
  username: string;
  email: string;
  avatarUrl: string | null;
  isAlreadyFriend: boolean | null;
  isRequestSentToThem: boolean | null;
  isRequestReceivedFromThem: boolean | null;
}

File: Backend/languaforge/src/main/java/com/tengelyhatalmak/languaforge/dto/UserAsFriendSearchResultDTO.java
package com.tengelyhatalmak.languaforge.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class UserAsFriendSearchResultDTO {
    private Integer id;
    private String username;
    private String email;
    private String avatarUrl;
}
